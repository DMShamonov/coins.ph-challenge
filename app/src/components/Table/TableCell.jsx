// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { TableCellPropsType } from './types.js.flow';

export default function TableCell({
  width,
  className,
  children,
}: TableCellPropsType): React.Node {
  return (
    <div className={classnames('table__cell', className)} style={{ width }}>
      {children}
    </div>
  );
}
