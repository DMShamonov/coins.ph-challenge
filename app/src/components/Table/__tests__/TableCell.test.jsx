import React from 'react';
import { shallow } from 'enzyme';

import TableCell from '../TableCell';

describe('`TableCell` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<TableCell />)).toMatchSnapshot();
  });

  test('with `children` prop', () => {
    expect(shallow(<TableCell>__CONTENT__</TableCell>)).toMatchSnapshot();
  });

  test('with `className` prop', () => {
    expect(shallow(<TableCell className="__CLASSNAME__">__CONTENT__</TableCell>)).toMatchSnapshot();
  });

  test('with `width` prop', () => {
    expect(shallow(<TableCell width="10%">__CONTENT__</TableCell>)).toMatchSnapshot();
  });
});
