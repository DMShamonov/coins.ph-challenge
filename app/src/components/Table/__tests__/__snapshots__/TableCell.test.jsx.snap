// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`TableCell\` component should render correctly with \`children\` prop 1`] = `
<div
  className="table__cell"
  style={
    Object {
      "width": undefined,
    }
  }
>
  __CONTENT__
</div>
`;

exports[`\`TableCell\` component should render correctly with \`className\` prop 1`] = `
<div
  className="table__cell __CLASSNAME__"
  style={
    Object {
      "width": undefined,
    }
  }
>
  __CONTENT__
</div>
`;

exports[`\`TableCell\` component should render correctly with \`width\` prop 1`] = `
<div
  className="table__cell"
  style={
    Object {
      "width": "10%",
    }
  }
>
  __CONTENT__
</div>
`;

exports[`\`TableCell\` component should render correctly with default props 1`] = `
<div
  className="table__cell"
  style={
    Object {
      "width": undefined,
    }
  }
/>
`;
